packages/jellyfish-testing/src/testing.ts

```diff
@@ -6,24 +6,31 @@ import { TestingRawTx } from './rawtx'
 import { TestingICX } from './icxorderbook'
 import { TestingMisc } from './misc'
 import { TestingGroupAnchor } from './anchor'
-import { ContainerGroup, MasterNodeRegTestContainer, StartOptions } from '@defichain/testcontainers'
+import { ContainerGroup, DeFiDContainer, MasterNodeRegTestContainer, StartOptions } from '@defichain/testcontainers'
 import { RegTestFoundationKeys } from '@defichain/jellyfish-network'
 import { JsonRpcClient } from '@defichain/jellyfish-api-jsonrpc'
 
 export class Testing {
+  public readonly container: MasterNodeRegTestContainer
   public readonly fixture = new TestingFixture(this)
-  public readonly token = new TestingToken(this.container, this.rpc)
-  public readonly poolpair = new TestingPoolPair(this.container, this.rpc)
-  public readonly rawtx = new TestingRawTx(this.container, this.rpc)
-  public readonly icxorderbook = new TestingICX(this)
-  public readonly misc = new TestingMisc(this.container, this.rpc)
+  public readonly token: TestingToken
+  public readonly poolpair: TestingPoolPair
+  public readonly rawtx: TestingRawTx
+  public readonly icxorderbook: TestingICX
+  public readonly misc: TestingMisc
```

Comment:
that's the issue and solution found out by @kodemon 
`Property xXx is used before its initialization`
preferably fix it via tsconfig, use es2020

Reply:
yea noticed it too before but it wasnt bugging me. now it is as im changing the constructor param abit. im now passing in a more generic DefiDContainer instead of `MasterNodeRegTestContainer` and forcibly cast it as `MasterNodeRegTestContainer` hence the change in the constructor

Reply:
Quick [reference](https://github.com/DeFiCh/jellyfish/blob/kodemon/whale-api-client-migration/tsconfig.json#L10) to change in whale migration used to solve this issue.